routing/blinding_test.go

```diff
@@ -172,11 +193,22 @@ func TestBlindedPaymentToHints(t *testing.T) {
 		// We can't assert that our functions are equal, so we check
 		// their output and then mark as nil so that we can use
 		// require.Equal for all our other fields.
-		require.Equal(t, expectedHint[0].ToNodePubKey(),
-			actualHint[0].ToNodePubKey())
+		require.Equal(t, expectedHint[0].EdgePolicy().ToNodePubKey(),
+			actualHint[0].EdgePolicy().ToNodePubKey())
+
+		actualHint[0].EdgePolicy().ToNodePubKey = nil
```

Comment:
function signatures cannot be compared because they will have a different address, so we nil them compare before that the functions produce the same `ToNodePubKey` and `nil` them after so that we can compare the hints at the end:

`require.Equal(t, expectedHint[0], actualHint[0])`

otherwise this would always fail.

Reply:
Cool thanks for the explanation! Could we add the docs here then?